src/components/Api.js

```diff
@@ -0,0 +1,90 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _request(url, options) {
+    return fetch(url, options)
+      .then(this._checkResponse)
+      .then((res) => res.json())
+      .catch(this._handleError);
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`   after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.